Src/main.c

```diff
@@ -86,6 +94,11 @@ static char	*PreArg[64];
 static int	PreCnt = 0;
 static char	out1[64];
 
+void pre_arg_append(char* arg) {
+  assert(PreCnt < sizeof(PreArg));
+  PreArg[PreCnt++] = arg;
+}
+
```

Comment:
This fixes the previous indexing that used `PreArg[++PreCnt]` in several places, which I assumed was incorrect, because it makes the array 1-indexed, leaving the first argument unused